### PR TITLE
[490] Import wave 1 RB contacts from Computacenter CSV file

### PR DESCRIPTION
**Pull request by aldavidson**
### Context

We have some spreadsheets from Computacenter with the contacts they hold for each of their accounts from wave 1.
We need to import those contacts into the DB as users, but also make sure they're linked to the right Responsible Bodies.
The task is made more complex by the fact that the source data we have, uses Computacenter references & GIAS UIDs to link things together.

### Changes proposed in this pull request

* Add db fields to store GIAS Id, GIAS Group UID, and computacenter_reference
* Add a new `ImportComputacenterReferencesService` to import Computacenter's 'SoldTo' (RB account ID) and `ShipTo` (school as shipping location) references to our schools & responsible_bodies
* Add a new `ImportLocalAuthorityGiasIdsService` to import GIAS Ids for Local Authorities from @benilovj 's manually-constructed CSV file
* Add  a new `ImportResponsibleBodyUsersFromComputacenterCsvService` to import the contacts
* Add rake task wrappers for all the above importers

### Guidance to review

You'll need to run the importers in this order:

```bash
# re-run existing importer to pick up the GIAS Group UIDs for trusts:
rake import:trusts
rake import:local_authority_gias_ids 
rake import:school_and_rb_computacenter_references CC_REFERENCES_FILE_URI=(file path or URI)
rake import:rb_users_from_cc_csv RB_USERS_CSV_URL=(file path or URI)
```

Ask @aldavidson or @benilovj for the data files if you need to run those last 2 cmds

### CODE REVIEW

File: app/models/responsible_body.rb [hunk 1]
```diff
@@ -32,4 +32,27 @@ def who_will_order_devices_label
       humanized_type.capitalize
     end
   end
+
+  def self.find_by_computacenter_urn!(cc_urn)
```
Comment (benilovj) [inline app/models/responsible_body.rb:36]: I think we should pull this and the subsequent methods into a module and include it into this class, to group these and make it explicit that these are standalone and are related to CC.

File: app/services/import_responsible_body_users_from_computacenter_csv_service.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
+require 'open-uri'
+require 'csv'
+
+class ImportResponsibleBodyUsersFromComputacenterCsvService
```
Comment (benilovj) [inline app/services/import_responsible_body_users_from_computacenter_csv_service.rb:4]: Not a fan of this class name or that it's a service. Could this be a `ComputacenterCsvFile` model with a `import_responsible_bodies` method?

File: db/schema.rb [hunk 3]
```diff
@@ -77,10 +77,10 @@
     t.datetime "created_at", precision: 6, null: false
     t.datetime "updated_at", precision: 6, null: false
     t.string "status", null: false
-    t.bigint "school_contact_id"
-    t.string "will_need_chromebooks"
     t.string "school_or_rb_domain"
     t.string "recovery_email_address"
+    t.bigint "school_contact_id"
```
Comment (benilovj) [inline db/schema.rb:82]: @aldavidson you mind re-creating your DB so this noise goes away?

**Review by benilovj — COMMENTED** [review]
Looking good so far, but it'd be really good to change the services to models that are based around files – this would make it clearer to understand at a glance IMHO.

File: app/models/responsible_body.rb (hunk 1 above)
Comment (aldavidson) [reply app/models/responsible_body.rb:36]: Yeah, I originally tried having these in a standalone class in the `Computacenter::` namespace, but it didn't quite work - a module may work better, will give it a try

File: app/services/import_responsible_body_users_from_computacenter_csv_service.rb (hunk 2 above)
Comment (aldavidson) [reply app/services/import_responsible_body_users_from_computacenter_csv_service.rb:4]: It was for consistency with the rest of the `ImportXYZService` classes - there's a Tech Debt ticket in the backlog to review our usage of the `Service` concept, it's got very muddled.

File: db/schema.rb (hunk 3 above)
Comment (aldavidson) [reply db/schema.rb:82]: i'll fix it

**Review by tonyheadford — APPROVED** [review]
LGTM!

File: app/models/responsible_body.rb (hunk 1 above)
Comment (aldavidson) [reply app/models/responsible_body.rb]: Done, that makes it cleaner 👍

File: db/schema.rb (hunk 3 above)
Comment (aldavidson) [reply db/schema.rb]: fixed

**Comment by aldavidson** [comment]
> Looking good so far, but it'd be really good to change the services to models that are based around files – this would make it clearer to understand at a glance IMHO.

I agree, just have not had chance to do that - I suggest we do that as a separate PR given the timescales we're working to

**Comment by benilovj** [comment]
> > Looking good so far, but it'd be really good to change the services to models that are based around files – this would make it clearer to understand at a glance IMHO.
> 
> I agree, just have not had chance to do that - I suggest we do that as a separate PR given the timescales we're working to

Yep, agreed it should be a different PR. @tonyheadford has introduced some really nice abstractions that will help.

**Pull request merged**